pkg/instancegroups/rollingupdate.go

```diff
@@ -135,17 +142,37 @@ func (c *RollingUpdateCluster) RollingUpdate(groups map[string]*cloudinstances.C
 					err = g.RollingUpdate(c, cluster, instanceGroups, false, c.MasterInterval, c.ValidationTimeout)
 				}
 
+				if err != nil {
+					// Remove function panic errors if an actual error occurred, otherwise that would be displayed
+					for k := range masterGroups {
+						resultsMutex.Lock()
+						results[k] = nil
+						resultsMutex.Unlock()
+					}
+				}
+
 				resultsMutex.Lock()
 				results[k] = err
 				resultsMutex.Unlock()
 
-				// TODO: Bail on error?
+				if err != nil {
+					// Stop before all masters are updated if an error occurs,
+					// to prevent more masters from failing and causing an outage
+					return
```

Comment:
also think a more verbose error is useful here

Reply:
Yes - I think the error is in `results[k]` so this is great.  Looking at this I don't think we need the wait group at all in this case, but we can tackle that cleanup later!